### PR TITLE
feat: OpenChannelList

### PR DESCRIPTION
**Pull request by HoonBaek**
## For Internal Contributors

[UIKIT-2086](https://sendbird.atlassian.net/browse/UIKIT-2086)

## Description Of Changes

* Create new smart components (modules)
  * CreateOpenChannel
  * OpenChannelList
* Add a renderHeader props into the ui/Modal component
* Add stringSet for OpenChannelLisit and CreateOpenChannel components
  * OPEN_CHANNEL_LIST__TITLE: 'Channels',
  * CREATE_OPEN_CHANNEL_LIST__TITLE: 'New channel profile',
  * CREATE_OPEN_CHANNEL_LIST__SUBTITLE__IMG_SECTION: 'Channel image',
  * CREATE_OPEN_CHANNEL_LIST__SUBTITLE__IMG_UPLOAD: 'Upload',
  * CREATE_OPEN_CHANNEL_LIST__SUBTITLE__TEXT_SECTION: 'Channel name',
  * CREATE_OPEN_CHANNEL_LIST__SUBTITLE__TEXT_PLACE_HOLDER: 'Enter channel name',
  * CREATE_OPEN_CHANNEL_LIST__SUBMIT: 'Create',

## Types Of Changes

What types of changes does your code introduce to this project?
Put an `x` in the boxes that apply_

- [ ] Bugfix
- [x] New feature
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance (ex) Prettier)
- [ ] Build configuration
- [ ] Improvement (refactor code)

### CODE REVIEW

File: src/smart-components/CreateOpenChannel/components/CreateOpenChannelUI/index.scss [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  position: relative;
+  display: inline-flex;
+  flex-direction: column;
+  justify-content: flex-start;
+
+  width: 100%;
+  height: 100%;
+  box-sizing: border-box;
+
+  .sendbird-create-open-channel-ui__profile-input__img-section {
```
Comment (sravan-s) [inline src/smart-components/CreateOpenChannel/components/CreateOpenChannelUI/index.scss:17]: Lets try not to go 3 levels in CSS

File: scripts/index_d_ts [hunk 2]
```diff
@@ -1716,6 +1716,7 @@ declare module '@sendbird/uikit-react/ui/Modal' {
     disabled?: boolean,
     type?: string,
     children: React.ReactElement | string,
+    renderHeader?: () => React.ReactElement,
   }
   const Modal: React.FC<ModalProps>;
```
Comment (sravan-s) [inline scripts/index_d_ts:1721]: Should add type defns for components

File: src/smart-components/CreateOpenChannel/components/CreateOpenChannelUI/index.scss [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      display: inline-flex;
+    }
+  }
+  .sendbird-create-open-channel-ui__profile-input__name-section {
+    position: relative;
+    margin-top: 8px;
+    display: inline-flex;
+    flex-direction: row;
+
+    .sendbird-create-open-channel-ui__profile-input__name-section__input {
```
Comment (sravan-s) [inline src/smart-components/CreateOpenChannel/components/CreateOpenChannelUI/index.scss:61]: Lets not have empty rules

File: src/smart-components/CreateOpenChannel/components/CreateOpenChannelUI/index.scss [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        justify-content: center;
+        width: 80px;
+        height: 80px; 
+        border-radius: 50%;
+        @include themed() {
+          background-color: t(bg-2);
+        }
+      }
+    }
+    .sendbird-create-open-channel-ui__profile-input__img-section__input {
```
Comment (sravan-s) [inline src/smart-components/CreateOpenChannel/components/CreateOpenChannelUI/index.scss:46]: Lets not have empty rules

File: src/smart-components/CreateOpenChannel/components/CreateOpenChannelUI/index.tsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  return (
+    <div className="sendbird-create-open-channel-ui">
+      <Modal
+        titleText={stringSet.CREATE_OPEN_CHANNEL_LIST__TITLE}
+        submitText={stringSet.CREATE_OPEN_CHANNEL_LIST__SUBMIT}
+        type={ButtonTypes.PRIMARY}
+        onCancel={closeModal}
+        renderHeader={renderHeader}
+        onSubmit={() => {
+          const channelName = inputTextRef?.current?.value;
```
Comment (sravan-s) [inline src/smart-components/CreateOpenChannel/components/CreateOpenChannelUI/index.tsx:44]: https://twitter.com/DavidKPiano/status/1569361453928300545?s=20&t=CJXyhVXgINPA5YVv2Rkqdw -> nit - do  want to try this method?

File: src/smart-components/OpenChannelList/components/OpenChannelListUI/index.scss [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    display: inline-flex;
+    flex-direction: column;
+
+    width: 100%;
+    height: calc(100% - 64px);
+    box-sizing: border-box;
+
+    overflow-y: scroll;
+
+    .sendbird-open-channel-list-ui__channel-list__item {}
```
Comment (sravan-s) [inline src/smart-components/OpenChannelList/components/OpenChannelListUI/index.scss:68]: same as before

File: src/smart-components/CreateOpenChannel/components/CreateOpenChannelUI/index.scss (hunk 1 above)
Comment (HoonBaek) [reply src/smart-components/CreateOpenChannel/components/CreateOpenChannelUI/index.scss:17]: let's refactor later

File: src/smart-components/CreateOpenChannel/components/CreateOpenChannelUI/index.tsx (hunk 5 above)
Comment (HoonBaek) [reply src/smart-components/CreateOpenChannel/components/CreateOpenChannelUI/index.tsx:44]: yes, I will try in the next refactoring step

File: scripts/index_d_ts [hunk 7]
```diff
@@ -667,6 +667,87 @@ declare module "SendbirdUIKitGlobal" {
...
+  export interface OpenChannelListProviderProps {
+    className?: string;
+    children?: React.ReactElement;
+    queries?: { openChannelListQuery?: UserFilledOpenChannelListQuery };
+    onChannelSelected?: OnOpenChannelSelected;
+  }
+
+  export interface OpenChannelListUIProps {
+    renderHeader?: () => React.ReactElement;
+    renderChannelPreview?: (props: RenderChannelPreviewProps) => React.ReactElement;
```
Comment (sravan-s) [inline scripts/index_d_ts:696]: RenderChannelPreviewProps -> in this file returns groupChannel, no?

Comment (sravan-s) [reply scripts/index_d_ts:696]: better have `RenderOpenChannelPreviewProps `

File: src/lib/SendbirdState.tsx [hunk 8]
```diff
@@ -1,12 +1,9 @@
...
 import type { User } from '@sendbird/chat';
 import type { Member } from '@sendbird/chat/groupChannel';
+import { Dispatch } from 'react';
 
-export type CustomUseReducerDispatcher = ({
-  type: string,
-  /* eslint-disable @typescript-eslint/no-explicit-any*/
-  payload: any,
-}) => void;
+export type CustomUseReducerDispatcher = Dispatch<{ type: string, payload: any }>;
```
Comment (sravan-s) [inline src/lib/SendbirdState.tsx:6]: no need to `* eslint-disable @typescript-eslint/no-explicit-any*/ `? worked okay?

File: scripts/index_d_ts (hunk 7 above)
Comment (HoonBaek) [reply scripts/index_d_ts:696]: good catch

File: src/lib/SendbirdState.tsx (hunk 8 above)
Comment (HoonBaek) [reply src/lib/SendbirdState.tsx:6]: It works fine now

**Review by sravan-s — APPROVED** [review]

**Pull request merged**